draft-ietf-quic-transport.md

```diff
@@ -1438,6 +1438,31 @@ New transport parameters can be registered according to the rules in
 {{iana-transport-parameters}}.
 
 
+## Cryptographic Message Buffering
+
+Implementations need to maintain a buffer of CRYPTO data received out of order.
+Because there is no flow control of CRYPTO frames, an endpoint could
+potentially force its peer to buffer an unbounded amount of data.
+
+Implementations MUST support buffering at least 4096 bytes of data received in
+CRYPTO frames out of order. Endpoints MAY choose to allow more data to be
+buffered during the handshake. A larger limit during the handshake could allow
+for larger keys or credentials to be exchanged. An endpoint's buffer size does
+not need to remain constant during the life of the connection.
+
+Being unable to buffer CRYPTO frames during the handshake leads to a connection
+failure. If an endpoint's buffer is exceeded during the handshake, it can expand
+its buffer temporarily to complete the handshake. If an endpoint cannot expand
+its buffer, it MAY close the connection with a CRYPTO_BUFFER_EXCEEDED error
```

Comment:
```suggestion
its buffer, it MUST close the connection with a CRYPTO_BUFFER_EXCEEDED error
```

I don't see any other recourse, do you?

Reply:
I was thinking it could grow its buffer, but I've already preconditioned with it not doing that, so a MUST here makes sense.